inc/user.class.php

```diff
@@ -4518,38 +4518,60 @@ static function getUniquePersonalToken() {
     * @param $ID user ID
     * @param string $field the field storing the token
     *
+    * @deprecated 9.2 @see User::getAuthToken()
+    *
     * @return string token
    **/
    static function getToken($ID, $field = 'personal_token') {
       global $DB;
 
+      Toolbox::deprecated('getToken() method is deprecated');
       $user = new self();
       if ($user->getFromDB($ID)) {
-         if (!empty($user->fields[$field])) {
-            return $user->fields[$field];
-         }
-         $token = self::getUniqueToken($field);
-         $user->update(['id'             => $user->getID(),
-                             $field           => $token,
-                             $field . "_date" => $_SESSION['glpi_currenttime']]);
-         return $user->fields[$field];
+         return $user->getAuthToken($field);
       }
 
       return false;
    }
 
+   /**
+    * Get token of a user. If it does not exists  then generate it.
+    *
+    * @param string $field the field storing the token
+    *
+    * @return string|false token or false in case of error
+    */
+   public function getAuthToken($field = 'personal_token') {
```

Comment:
I do not know which version is targetted, but I think a `@since` tag should be added here.